### PR TITLE
libslab: Fix compilation warnings and replace deprecated functions

### PR DESCRIPTION
**Pull request by zhuyaliang**

### CODE REVIEW

**Review by lukefromdc — APPROVED** [review]
Builds and runs fine, again mostly  reimplementation of deprecated GTK3 functions and a couple replacements with newer/supported ones. Looks good to me

File: libslab/app-resizer.c [hunk 1]
```diff
@@ -275,11 +275,19 @@ static gboolean
...
 
 	GtkAllocation widget_allocation;
 	gtk_widget_get_allocation (widget, &widget_allocation);
 
-	gdk_cairo_set_source_color (cr, gtk_widget_get_style (widget)->base);
+	context = gtk_widget_get_style_context (widget);
+	gtk_style_context_get (context,
+                         GTK_STATE_FLAG_NORMAL,
+                         "background-color", &bg_rgba,
+                         NULL);
```
Comment (raveit65) [inline libslab/app-resizer.c:288]: Intends
```suggestion
	gtk_style_context_get (context,
                               GTK_STATE_FLAG_NORMAL,
                               "background-color", &bg_rgba,
                               NULL);
```

File: libslab/app-resizer.c [hunk 2]
```diff
@@ -289,19 +297,25 @@ app_resizer_paint_window (GtkWidget * widget, cairo_t * cr, AppShellData * app_d
...
 		GtkAllocation selected_widget_allocation;
-
 		gtk_widget_get_allocation (selected_widget, &selected_widget_allocation);
 
-		gdk_cairo_set_source_color (cr, gtk_widget_get_style (selected_widget)->light);
-		cairo_set_line_width(cr, 1);
+		gtk_style_context_get (context,
+                               GTK_STATE_FLAG_PRELIGHT,
+                               "background-color", &rgba,
+                               NULL);
```
Comment (raveit65) [inline libslab/app-resizer.c:307]: Indents
```suggestion
		gtk_style_context_get (context,
                                       GTK_STATE_FLAG_PRELIGHT,
                                       "background-color", &rgba,
                                       NULL);
```

File: libslab/application-tile.c [hunk 3]
```diff
@@ -417,19 +417,43 @@ create_header (const gchar *name)
 	return header;
 }
 
+static void
+set_background_color (GtkWidget *widget,
+                    GdkRGBA   *rgba)
```
Comment (raveit65) [inline libslab/application-tile.c:422]: Indents
```suggestion
set_background_color (GtkWidget *widget,
                      GdkRGBA   *rgba)
```

File: libslab/application-tile.c [hunk 4]
```diff
@@ -417,19 +417,43 @@ create_header (const gchar *name)
...
+	GdkRGBA         *rgba = NULL;
 
 	subheader = gtk_label_new (desc);
 	gtk_label_set_ellipsize (GTK_LABEL (subheader), PANGO_ELLIPSIZE_END);
 	gtk_label_set_xalign (GTK_LABEL (subheader), 0.0);
+	context = gtk_widget_get_style_context (subheader);
+	gtk_style_context_get (context,
+                           GTK_STATE_FLAG_INSENSITIVE,
+                           "background-color", &rgba,
+                           NULL);
```
Comment (raveit65) [inline libslab/application-tile.c:454]: Indents
```suggestion
	gtk_style_context_get (context,
                               GTK_STATE_FLAG_INSENSITIVE,
                               "background-color", &rgba,
                               NULL);
```

File: libslab/slab-section.c [hunk 5]
```diff
@@ -45,22 +45,60 @@ slab_section_finalize (GObject * obj)
...
+
 	switch (SLAB_SECTION (widget)->style)
 	{
 	case Style1:
-		gtk_widget_modify_fg (SLAB_SECTION (widget)->title, GTK_STATE_NORMAL,
-			&gtk_widget_get_style (widget)->bg[GTK_STATE_SELECTED]);
+		gtk_style_context_get (context,
+                               GTK_STATE_FLAG_SELECTED,
+                               "background-color", &rgba,
+                               NULL);
```
Comment (raveit65) [inline libslab/slab-section.c:82]: Indents
```suggestion
		gtk_style_context_get (context,
                                       GTK_STATE_FLAG_SELECTED,
                                       "background-color", &rgba,
                                       NULL);
```

File: libslab/slab-section.c [hunk 6]
```diff
@@ -45,22 +45,60 @@ slab_section_finalize (GObject * obj)
...
 		break;
 	case Style2:
 		if (SLAB_SECTION (widget)->selected)
-			gtk_widget_modify_fg (SLAB_SECTION (widget)->title, GTK_STATE_NORMAL,
-				&gtk_widget_get_style (widget)->dark[GTK_STATE_SELECTED]);
+		{
+			gtk_style_context_get (context,
+                                   GTK_STATE_FLAG_SELECTED,
+                                   "background-color", &rgba,
+                                   NULL);
```
Comment (raveit65) [inline libslab/slab-section.c:91]: Indents
```suggestion
			gtk_style_context_get (context,
                                               GTK_STATE_FLAG_SELECTED,
                                               "background-color", &rgba,
                                               NULL);
```

File: libslab/slab-section.c [hunk 7]
```diff
@@ -45,22 +45,60 @@ slab_section_finalize (GObject * obj)
...
+			set_override_color (SLAB_SECTION (widget)->title, rgba);
+		}
 		else
-			gtk_widget_modify_fg (SLAB_SECTION (widget)->title, GTK_STATE_NORMAL,
-				&gtk_widget_get_style (widget)->text[GTK_STATE_INSENSITIVE]);
+		{
+			gtk_style_context_get (context,
+                                   GTK_STATE_FLAG_INSENSITIVE,
+                                   "color", &rgba,
+                                   NULL);
```
Comment (raveit65) [inline libslab/slab-section.c:99]: Indents
```suggestion
			gtk_style_context_get (context,
                                               GTK_STATE_FLAG_INSENSITIVE,
                                               "color", &rgba,
                                               NULL);
```

**Comment by raveit65** [comment]
Beside from formatting issues m-c-c main window works fine and looks like before.

File: libslab/app-resizer.c (hunk 1 above)
Comment (zhuyaliang) [reply libslab/app-resizer.c]: done

File: libslab/application-tile.c (hunk 3 above)
Comment (zhuyaliang) [reply libslab/application-tile.c]: done

File: libslab/application-tile.c (hunk 4 above)
Comment (zhuyaliang) [reply libslab/application-tile.c]: done

**Review by raveit65 — APPROVED** [review]
LGTM

**Pull request merged**